src/FsUnit.MsTestUnit/FsUnit.fs

```diff
@@ -1,119 +1,117 @@
-﻿module FsUnit.MsTest
-
-open System
-open Microsoft.VisualStudio.TestTools.UnitTesting
-open NHamcrest
-open NHamcrest.Core
-open System.Collections
-
-let inline private assertThat(actual, matcher:IMatcher<'a>) =
-    if not (matcher.Matches(actual)) then
-            let description = new StringDescription()
-            matcher.DescribeTo(description)
-            let mismatchDescription = new StringDescription()
-            matcher.DescribeMismatch(actual, mismatchDescription)
-            raise (new AssertFailedException(sprintf "%s %s" (description.ToString()) (mismatchDescription.ToString())))
-
-#if NETSTANDARD1_6
-type Assert with
-    static member That'<'a> (actual, matcher:IMatcher<'a>) =
-        assertThat(actual, matcher)
-#else
-type Assert with
-    static member That<'a> (actual, matcher:IMatcher<'a>) =
-        assertThat(actual, matcher)
-#endif
-
-let inline should (f : 'a -> ^b) x (y : obj) =
-    let c = f x
-    let y =
-        match y with
-        | :? (unit -> unit) as assertFunc -> box assertFunc
-        | _ -> y
-    if box c = null then
+﻿module FsUnit.MsTest
+
+open System
+open Microsoft.VisualStudio.TestTools.UnitTesting
+open NHamcrest
+open NHamcrest.Core
+open System.Collections
+
+let inline private assertThat(actual, matcher:IMatcher<'a>) =
+    if not (matcher.Matches(actual)) then
+            let description = new StringDescription()
+            matcher.DescribeTo(description)
+            raise (new AssertFailedException(sprintf "%s %s" (description.ToString()) (sprintf "was %A" actual)))

```

Comment:
Are you sure that this change does not break other mismatch descriptions?
Looks like `matcher.DescribeMismatch` was intended to build 'good' error messages provided by frameworks, but you replace `mismatchDescription.ToString()` by `sprintf "was %A" actual` 

How it affect error messages for not pure equal operators like `haveLength`, `greaterThan`, `startWith` and `subsetOf`?

Reply:
Yes, I'm sure. I wrote tests as seen in the next commit.

`matcher.DescribeMismatch` breaks the generic types in F# as seen in the opening comment the type `Result.Ok` will be `Microsoft.FSharp.Core.FSharpResult`2[System.String,System.Object]` in the "failing" message in FsUnit.

This will not affect the other operators, I checked this with tests.

Reply:
Here we could also take another string in the Exception.Message like `sprintf "Expected: %s \nActual: %A" (description.ToString()) actual` which will printed similar to the Xunit part.
Nunit prints something like "Expected: %s \r\n   But was:   %s".
I choosed the "was %A", because this string was returned by NHamcrest.